service/history/workflow/mutable_state_impl.go

```diff
@@ -4920,14 +4919,16 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionOptionsUpdatedEvent(event *his
 		}
 	}
 
+	// Finally, reschedule the pending workflow task if so requested.
+	if requestReschedulePendingWorkflowTask {
+		return ms.reschedulePendingWorkflowTask(true)
+	}
 	return nil
 }
 
 func (ms *MutableStateImpl) updateVersioningOverride(
 	override *workflowpb.VersioningOverride,
-) error {
-	previousEffectiveDeployment := ms.GetEffectiveDeployment()
-	previousEffectiveVersioningBehavior := ms.GetEffectiveVersioningBehavior()
+) (requestReschedulePendingWorkflowTask bool, err error) {
```

Comment:
nit: Prefer avoiding named return values in large and complex functions. It hurts readability since return statements are naked and hard to know what values are being returned.

Reply:
Using `var` now, looks fine 👍